docs/developer/develop-vscode.md

```diff
@@ -0,0 +1,135 @@
+
+## Before you start
+*   Check out the [How it Works](../user/how-it-works.md) page to see how the different
+    components work within Turbinia.
+*   Make sure to follow the Turbinia
+    [developer contribution guide](contributing.md).
+
+#### Step 1 - Install required software
+Prepare your OS:
+* Install Visual Studio Code and install the Remote Development extension pack.
+* Install Docker on your operating system (eg Docker Desktop on OSX)
+
+#### Step 2 - Fork Turbinia
+Fork Turbinia on Github and create a new feature branch to work on.
+```
+$ git clone https://github.com/[your-github-user-id]/turbinia.git
+$ git remote add upstream https://github.com/google/turbinia.git
+$ git checkout -b my-new-feature
+$ cd turbinia
+```
+
+#### Step 3 - Open in Visual Studio Code
+Open the folder in vscode and choose to "Reopen in Container" when asked (vscode will see the ```.devcontainer``` folder in the turbinia cloned source tree). Vscode will build your Turbinia development container and that will take a couple of minutes.
+
+Note: if vscode does not ask you to reopen in a container you need to verify you have installed the Remote Development extension!
+
+When this is finished continue with Step 4 for a local Turbinia setup or Step 6 for a GCP hybrid setup. 
+
+#### Step 4 - Local Turbinia setup
+The local turbinia setup will use redis and celery. Let's create the configuration file for this setup.
+```
+$ sed -f ./docker/vscode/redis-config.sed ./turbinia/config/turbinia_config_tmpl.py > ~/.turbiniarc
+```
+
+#### Step 5 - Local Turbinia verification
+Let's verify that the local setup is working before we start developing and debugging. We are going to start a server and worker in seperate vscode terminals and create a Turbinia request in a third.. Open up 3 vscode terminals and execute below commands.
+
+Start server
+```
+$ python3 turbinia/turbiniactl.py -S server
+```
+Start worker
+```
+$ python3 turbinia/turbiniactl.py -S celeryworker
+```
+Fetch and process some evidence
+```
+$ curl https://raw.githubusercontent.com/obsidianforensics/hindsight/master/tests/fixtures/profiles/60/History > History
+$ tar -vzcf /evidence/history.tgz History
+$ python3 turbinia/turbiniactl.py compresseddirectory -l /evidence/history.tgz 
+$ python3 turbinia/turbiniactl.py status -r [request_id]
+```
+
+This should process the evidence and show output in each terminal for server and worker. Results will be stored in ```/evidence```.
+
+You're done :). Go to Step 8 for a vscode debugging example.
+
+#### Step 6 - GCP hybrid Turbinia setup
+Follow the ‘GCP Setup’ section [here](https://turbinia.readthedocs.io/en/latest/user/install-manual.html) and setup Cloud Functions, a GCE bucket, Datastore and PubSub.
+* Create a pubsub topic ,eg ‘turbinia-dev’.
+* Create a GCE storage bucket with a unique name.
+* Create the GCP Cloud Functions
+```
+$ git clone https://github.com/forseti-security/osdfir-infrastructure.git
```

Comment:
If this is only being used for the cloud function files, we should be able to take them out of the `turbinia/tools/gcf_init` in the Turbinia project/code.   Also, if you do the config setup first, you can replace all of this with a call to `turbinia/tools/gcf_init/deploy_gcf.py` which reads the config and does these same gcloud deployment commands.